src/Panel.js

```diff
@@ -94,9 +94,8 @@ class Panel extends React.Component {
   getExpanded() {
     const { eventKey } = this.props;
     const { activeKey } = this.context.$bs_panelGroup || {};
-
-    return this.props.expanded != null || activeKey === undefined
```

Comment:
Shouldn't `this.props.expanded` take precedence over `activeKey` though? I think the bug is more in `<PanelGroup>`.

Reply:
If `<PanelGroup>` is not an `accordion` then `activeKey` is not on the `$bs_panelGroup` context  and none of the children controls like `<Panel>` even have `activeKey` available.

It took me a good hour of instrumenting code with log statements to figure out the first point above, the code looks simple but I took me a while to get the pattern in my head then think about it.

This then lead me to `<Panel>` really does not care about its expanded status at all if `activeKey` is present. This works out that if `activeKey` changes via `<PanelGroup>.handleSelect()` then all children need to decide if they are then open or closed purely based on `activeKey`.

Please feel free to point out holes in my logic.

Reply:
I see, so this is the Uncontrollable thing. @jquense is there a way to distinguish between an `expanded` coming down from Uncontrollable v. an explicit one?

Reply:
nope :/

Reply:
@taion is there specific condition you are thinking of that you have in mind that requires knowing about source of expanded being Uncontrollabe or explicit for some behaviour ?

Reply:
Could it come up that someone will do something like:

```js
<PanelGroup accordion>
  <Panel expanded>{...}</Panel>
  <Panel>{...}</Panel>
  <Panel>{...}</Panel>
</PanelGroup>
```

Or what should we do in that case? Warn, I guess?

Reply:
Do you have a development vs production model warning similar to React ? 
If so that would seem easy enough to warn about ?

Reply:
Right and we do something like that for tabs, so it would make sense to do the same thing here.

Reply:
The basic check would be in constructor of Panel throw Error if expanded is set true or false (not undefined) if there is a `$bs_panelGroup.activeTab` present on context.

This type of parameter checking Error I see raised in a dozen or so places in the code base all seem to not check for production or development so they will still be active in production builds unlike many of React's notifications.

eg. https://github.com/react-bootstrap/react-bootstrap/blob/master/src/TabContainer.js#L21-L27

Reply:
While checking behaviours, i just found that a `<Panel>` with `expanded` but no `onToggle` will get an error generated by uncontrollable as follows.

```
ERROR: 'Warning: Failed prop type: You have provided a `expanded` prop to `Panel` 
without an `onToggle` handler. This will render a read-only field. If the 
field should be mutable use `defaultExpanded`. Otherwise, set `onToggle`
in Uncontrolled(Panel)'
```
This error is great in general but in our case is not help this case  as providing `expanded` (without `onToggle`) to a `Panel` inside a `PanelGroup` of type `accordion` would generate this error, but the developer is just as likely to add `onToggle` to address the error as remove `expanded`, now given the current bug fix this won't stop the Panel from behaving as a member of and accordion panel group I believe, but its dead code that currently does nothing and may cause harm in the future if implementation of controls changes. If the developer is then relying on something in the panels `onToggle` he may get hard to understand behaviour.

After an initial attempt at detecting if a Panel inside an Accordion is being passed `expanded` or `onToggle` to generate an Error. I believe I can't do this without knowing if uncontrolled is managing the `expanded` and `onToggle` properties or not as @taion raised earlier.

Because `uncontrollable` provides these properties if the user does not then I can't just see if the props `expanded` or `onToggle` are provided to a `Panel` with a accordion panel group parent.

Reply:
Yeah. That's fine. Just emit a `warning` if `expanded` is not nully but `activeKey` is present?

Reply:
I am not sure if this is feasible.

Could I map over the children provided to PanelGroup with accordion enabled and emit an Error from that side if there is expanded onToggle or defaultExpanded explicitly set ?

Reply:
It's just a warning. No harm emitting a second warning.

Reply:
So far in my initial attempts the second one does not get emitted.
I will look at his again.